pootle/static/js/browser/components/BrowserTable.js

```diff
@@ -0,0 +1,328 @@
+            // quick reverse
+            s.reverseSort = !s.reverseSort;
+            this.sortedKeys = this.sortedKeys.reverse();
+        } else {
+            s.sortColumn = colN;
+            s.reverseSort = false;
+            this.sortedKeys = this.sortedKeys.sort(this.sortFunc);
+        }
+        this.setState(s);
+    },
+
+    getClassName(colN) {
+        if (colN == this.state.sortColumn) {
+            return this.state.reverseSort ? 'reverse' : 'normal'
+        }
+    },
+
+    createRow(key, index) {
+        var i = this.props.items[key];
+
+        if (i.is_disabled && !this.state.showDisabledRows) {
+            return;
+        }
+
+        const trClasses = cx('item', {
+            'is-visible': true, // TODO: remove?
+            'is-disabled': i.is_disabled,
+            'is-dirty': i.is_dirty,
+        });
+
+        var icon = 'icon-file';
+        if (i.treeitem_type == 1) icon = 'icon-folder';
+        if (i.treeitem_type == 2) icon = 'icon-project';
+
+        return (
+            <tr key={key} className={trClasses}>
+                <td className="stats-name">
+                    <a href={i.pootle_path} title={i.title}><i className={icon} />{i.title}</a>
+                </td>
+                <td className="stats-graph"><ProgressBar
+                    total={i.total}
+                    fuzzy={i.fuzzy}
+                    translated={i.translated}
+                /></td>
+                <td className="stats-number total">
+                    <a href={i.translate_url} className="stats-data">
+                        <ColoredNumber n={i.total} />
+                    </a>
+                </td>
+                <td className="stats-number last-updated">
+                    <TimeSince timestamp={i.lastupdated} />
+                </td>
+                <td className="stats-number critical">
+                    <NumberPill
+                        n={i.critical}
+                        url={i.translate_url + '#filter=checks&category=critical'}
+                    />
+                </td>
+                <td className="stats-number suggestions">
+                    <NumberPill
+                        n={i.suggestions}
+                        url={i.translate_url + '#filter=suggestions'}
+                    />
+                </td>
+                <td className="stats-number need-translation">
+                    <NumberPill
+                        n={i.total - i.translated}
+                        url={i.translate_url + '#filter=incomplete'}
+                    />
+                </td>
+                <td className="last-activity">
+                    <LastActivity {...i.lastaction} />
+                </td>
+            </tr>
+        );
+    },
+
+    render() {
+        if (!this.sortedKeys.length) {
+            // TODO: provide styling for the message
+            return (
+                <noscript/>
+            );
+        }
+
+        var toggle = '';
```

Comment:
Review if this is still needed or rather one can use `null` or leave it as `undefined`.